### PR TITLE
Updated pyright to latest version

### PR DESCRIPTION
**Pull request by erictraut**
Fixed a few imports in "cryptography" package that reference symbols that are not re-exported and therefore trigger the "reportPrivateImportUsage" check in newer versions of pyright.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Thanks! Do you know what's going on with the annotations in unchanged files (https://github.com/python/typeshed/pull/7132/files#file-stdlib-binascii-pyi-L6)?

**Comment by erictraut** [comment]
I'm looking into that now. I'm not able to repro it locally.

**Comment by erictraut** [comment]
I just noticed that the pre-commit hook automatically fixed the isort violation. That's really cool!

**Comment by erictraut** [comment]
I'm mystified by these CI errors. I'm not able to repro this locally, and I don't see any pattern. I guess I'll try to binary search based on pyright versions to see if that gives me a clue.

**Comment by erictraut** [comment]
I found the cause of the mystery CI errors. It's caused by a regression that I introduced in pyright 1.1.207. Surprisingly, I've released 10 versions since then, and no one has reported the problem. It's likely that the combination of features and configuration options is unique to typeshed. I'll fix the bug in the next version of pyright (1.1.218). In the interim, I recommend merging this PR as it stands. It pins pyright to version 1.1.206, which predates the introduction of this regression.

**Comment by JelleZijlstra** [comment]
Thanks for figuring out the bug! Looks like you published 218 already, so we might as well update to it. I will edit the PR.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
         python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "3.11"]
       fail-fast: false
     env:
-      PYRIGHT_VERSION: 1.1.204 # Must match pyright_test.py.
+      PYRIGHT_VERSION: 1.1.206 # Must match pyright_test.py.
```
Comment (JelleZijlstra) [inline .github/workflows/tests.yml:75]:
```suggestion
      PYRIGHT_VERSION: 1.1.218 # Must match pyright_test.py.
```

File: tests/pyright_test.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import sys
 from pathlib import Path
 
-_PYRIGHT_VERSION = "1.1.204"  # Must match .github/workflows/tests.yml.
+_PYRIGHT_VERSION = "1.1.206"  # Must match .github/workflows/tests.yml.
```
Comment (JelleZijlstra) [inline tests/pyright_test.py:8]:
```suggestion
_PYRIGHT_VERSION = "1.1.218"  # Must match .github/workflows/tests.yml.
```

**Comment by JelleZijlstra** [comment]
Also, I noticed that your release announcement (https://github.com/microsoft/pyright/releases/tag/1.1.218) is titles "1.1.128" instead of "1.1.218". The tag itself is right though.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**